typescript/src/bridge.ts

```diff
@@ -4,6 +4,16 @@ import {
   DecomposedRawTransaction,
 } from "./bitcoin"
 
+import { BigNumberish, BigNumber } from "ethers"
+
+export interface PendingRedemption {
+  redeemer: string
+  requestedAmount: BigNumber
+  treasuryFee: BigNumber
+  txMaxFee: BigNumber
+  requestedAt: number
+}
```

Comment:
We can do that, but notice these two interfaces are not exactly the same:
* `PendingRedemption` from `bridge.ts` is exactly what is returned from the `Bridge` contract (including field names).  It is unprocessed. The  field `txMaxFee` is the maximum fee share for that particular pending redemption.

* `RedemptionRequest` from `redemption.ts`  contains already processed data that `createRedemptionTransaction` needs - the `feeShare` is the actual fee share that the will be used which does not necessarily equal `txMaxfree` (it can be smaller, for now I just copy `txMaxfree` to `feeShare`, I left a TODO how we can handle it).

Reply:
Yes and this is wrong because we tie ourselves to the `Bridge` specific implementation, not their interface. If we do that, the current ETH Bridge implementation details will leak into our library thus making the potential cross-chain support much harder.  At this point, we do not need to care what exactly is returned from the Ethereum `Bridge` contract. We must care about building a consistent universal model of all "entities". This is why I propose to stick with `RedemptionRequest` defined in `redemption.ts`. This is our core type representing a redemption request. All implementation-specific flavors will live separately. For example, we could have an `ethereum.ts` file containing a `Bridge` implementation that would use a contract binding and use some internal type to represent a redemption request.

Reply:
OK, it makes sense. Done in 5c98ec9c19f2a3d2681e738493650312002d4c12.
We now use [a single interface](https://github.com/keep-network/tbtc-v2/blob/4712bc31a42f6c172ee6657eecd607781616cfb2/typescript/src/redemption.ts#L17) to represent the redemption returned from the Bridge, as well as the input data to `createRedemptionTransaction` function.